backend/main.py

```
@@ -302,82 +330,58 @@ def create_spot(
+    image_url = _upload_image_from_base64(spot_update.image_base64, folder="spots") if spot_update.image_base64 else None
+    try:
+        updated_spot = crud.update_spot(
+            db,
+            spot_id,
+            current_user.id,
+            spot_update,
+            image_url=image_url
         )
-    )
+    except ValueError:
+        raise HTTPException(status_code=404, detail="Spot not found")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)